### DISCUSSION TITLE
How to stop built file hashes from randomly changing?

### DISCUSSION THREAD

**Discussion by fabis94** [nuxt/nuxt#27610] [Questions]
I'm trying to build the app and upload its sourcemaps in 2 different CI steps, but for some reason the app build builds different filenames in both scenarios, even tho the git commit (codebase version) is the same.

I even made sure that sourcemaps are built in both scenarios (cause apparently just toggling source maps changes the filename hashes, which is absurd), but still - the filename content hashes end up being different.

What controls the contenthash and how do I ensure it stays the same across builds in different environments? This is about the latest nuxt 3.x version

**Comment by fabis94** [comment]
The differences I have between the 2 CI step environments:
- Project is stored at different locations in the filesystem
- One has git, the other one might not
- One runs in a docker image based on a slim node image, the other runs in a default CircleCI worker environment

**Reply by TheAlexLichter** [reply]
With 3.12 it should give you a stable build when not changing content.
As soon as you change the content, the filenames should change too (to avoid caching old, outdated JS)

**Reply by fabis94** [reply]
@manniL You mention 3.12 specifically, was there some kind of bug or issue before where that was not the case?

Also, I'm inferring from your answer, that the build environment does not impact the file hashes in any way, and its only their contents that change it, right?

**Reply by TheAlexLichter** [reply]
Correct, https://github.com/nuxt/nuxt/pull/27258 in particular


It should, yes.
